src/itemset.js

```diff
@@ -203,30 +211,51 @@ define([
     }
 
     function itemsetWidget(mug, options) {
-        var widget = datasources.dataSourceWidget(mug, options, "Data Source"),
+        var widget = datasources.fixtureWidget(mug, options, "Data Source"),
             super_getUIElement = widget.getUIElement,
             super_getValue = widget.getValue,
             super_setValue = widget.setValue,
-            handleChange = widget.handleChange.bind(widget),
-            labelRef = refSelect("label_ref", "Choice Label", widget.isDisabled()),
-            valueRef = refSelect("value_ref", "Choice Value", widget.isDisabled());
+            super_handleChange = widget.handleChange.bind(widget),
+            labelRef = refSelect("label_ref", "Choice Label", false),
+            valueRef = refSelect("value_ref", "Choice Value", false),
+            filterRef = refSelect("filter_ref", "Choice Filter", false);
+
+        function updateAutoComplete() {
+            if (widget.getValue().instance) {
+                var sources = datasources.autocompleteChoices(widget.getValue().instance.src);
+                labelRef.addAutoComplete(sources);
+                valueRef.addAutoComplete(sources);
+            }
+        }
 
-        labelRef.onChange(handleChange);
-        valueRef.onChange(handleChange);
+        var local_handleChange = function() {
+            updateAutoComplete();
+            super_handleChange();
+        };
+
+        widget.handleChange = local_handleChange;
+
+        labelRef.onChange(super_handleChange);
+        valueRef.onChange(super_handleChange);
+        filterRef.onChange(super_handleChange);
 
         widget.getUIElement = function () {
             return super_getUIElement()
                 .append(labelRef.element)
-                .append(valueRef.element);
+                .append(valueRef.element)
+                .append(filterRef.element);
         };
 
         widget.getValue = function () {
-            var val = super_getValue();
+            var val = super_getValue(),
+                s = val.query.search(END_FILTER),
+                filter = s === -1 ? '' : val.query.slice(s);
             return {
                 instance: ($.trim(val.src) ? {id: val.id, src: val.src} : null),
-                nodeset: val.query,
+                nodeset: val.query ? val.query.replace(END_FILTER, '') : '',
                 labelRef: labelRef.val(),
-                valueRef: valueRef.val()
+                valueRef: valueRef.val(),
+                filterRef: filterRef.val() || filter
```

Comment:
This was due to an edge case. As it is now cases are:
1) select from dropdown, no `filterRef` value, only have query with no filtering
2) select from drop down, fill out `filterRef`, `fiilterRef` is tacked on to the end of the query
3) use advanced editor with no filter on the end, fill out `filterRef`, `filterRef` is tacked on to the end of the query
4) use advanced editor with a filter on the end, don't fill out `filterRef`, the filter is put into `filterRef` and the rest of the query is put into the dropdown
5) use advanced editor with a filter on the end, fill out `filterRef`, the `filterRef` overrides the filter you put in the advanced editor

Thinking about it now it seems that maybe that filter should not be erased, but should stay in the dropdown if `filterRef` is already filled out. The other cases seem good to go though